### PR TITLE
CSHARP-3068: Raise error when hint option is provided on unacknowledged writes against any server version

### PR DESCRIPTION
**Pull request by MikalaiMazurenka**
https://jira.mongodb.org/browse/CSHARP-3068
https://evergreen.mongodb.com/version/5ea7f670e3c33134ae4775ce

### CODE REVIEW

**Review by rstam — APPROVED** [review]
LGTM

File: tests/MongoDB.Driver.Core.Tests/Core/Operations/RetryableDeleteCommandOperationTests.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                exception.Should().BeNull();
+            }
+            else
+            {
+                exception.Should().BeOfType<MongoCommandException>();
+            }
+        }
+    }
+}
+
```
Comment (DmitryLukyanov) [inline tests/MongoDB.Driver.Core.Tests/Core/Operations/RetryableDeleteCommandOperationTests.cs]: empty line

**Review by DmitryLukyanov — APPROVED** [review]
Minor comment and then LGTM

**Review by vincentkam — APPROVED** [review]
LGTM! 😸

**Review by DmitryLukyanov — APPROVED** [review]
LGTM

File: tests/MongoDB.Driver.Core.Tests/Core/Operations/RetryableDeleteCommandOperationTests.cs (hunk 1 above)
Comment (MikalaiMazurenka) [reply tests/MongoDB.Driver.Core.Tests/Core/Operations/RetryableDeleteCommandOperationTests.cs]: Done.

**Pull request closed without merging**